### PR TITLE
IGNITE-16392 PageMemory data regions configuration; porting of PageMemoryNoStoreImpl and all other necessary components.

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-16392

### CODE REVIEW

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Value(hasDefault = true)
+    public long maxSize = DFLT_DATA_REGION_MAX_SIZE;
+
+    @ConfigValue
+    public MemoryAllocatorConfigurationSchema memoryAllocator;
+
+    @OneOf({"DISABLED", "RANDOM_LRU", "RANDOM_2_LRU"})
+    @Value(hasDefault = true)
+    public String evictionPolicy = "DISABLED";
```
Comment (tkalkirill) [inline modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java:60]: Can use constants?

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    @OneOf({"RANDOM_LRU", "SEGMENTED_LRU", "CLOCK"})
+    @Value(hasDefault = true)
+    public String replacementMode = "CLOCK";
```
Comment (tkalkirill) [inline modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java:64]: Can use constants?

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    @Value(hasDefault = true)
+    public int emptyPagesPoolSize = 100;
+
+    @Value(hasDefault = true)
+    public long checkpointPageBufSize = 0;
+
+    @Value(hasDefault = true)
+    public boolean lazyMemoryAllocation = true;
+
+    // Warmup.
```
Comment (tkalkirill) [inline modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java:78]: Maybe jira ticket?

File: modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationTestUtils.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.apache.ignite.configuration.ConfigurationTree;
+
+/**
+ * Utility methods, related to testing configurations.
+ */
+public class ConfigurationTestUtils {
+    /**
+     * Casts dynamic configuration to its specific type. Can be applied to polymorphic configuration instances.
+     */
+    public static  <C extends ConfigurationTree> C fixConfiguration(C cfg) {
```
Comment (tkalkirill) [inline modules/configuration/src/test/java/org/apache/ignite/internal/configuration/ConfigurationTestUtils.java:29]:
```suggestion
    public static <C extends ConfigurationTree> C fixConfiguration(C cfg) {
```

**Review by tkalkirill — APPROVED** [review]

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java (hunk 1 above)
Comment (ibessonov) [reply modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java:60]: Sure

File: modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java (hunk 3 above)
Comment (ibessonov) [reply modules/api/src/main/java/org/apache/ignite/configuration/schemas/store/PageMemoryDataRegionConfigurationSchema.java:78]: Next time, I'll remove the comment

**Pull request merged**